### PR TITLE
Update repository with recent discussion and begin prototypes

### PR DESCRIPTION
**Pull request by starksm64**

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/jwt/JWTCallerPrincipal.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A standard CallerPrincipal implementation that provides access to the JWT claims that are required by
+ * the microprofile token.
+ */
+public class JWTCallerPrincipal extends CallerPrincipal {
+    /** TODO: need to decide on a JWT parsing/verification library or create one */
+    private JsonWebToken jwt;
```
Comment (johnament) [inline api/src/main/java/org/eclipse/microprofile/jwt/JWTCallerPrincipal.java]: Each impl needs to decide that, not the spec.  The spec may require a base interface to be satisfied, but please avoid using Keycloak in the spec.

File: api/src/main/resources/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (johnament) [inline api/src/main/resources/LICENSE]: We should only need a root LICENSE.

File: api/src/test/resources/jwk-content1.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  "exp": 1311281970,
+  "iat": 1311280970,
+  "auth_time": 1311280969,
+  "roles": [
+    "role-in-realm",
+    "user",
+    "manager"
+  ],
+  "groups": ["group1", "group2"],
+  "resource_access": {
```
Comment (johnament) [inline api/src/test/resources/jwk-content1.json]: Would a consumer of this specifically look for their service, and add proper groups/roles based on that?  How would a developer specify which if their service?

File: tck/pom.xml [hunk 4]
```diff
@@ -49,36 +49,16 @@
...
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api</artifactId>
+            <version>${project.version}</version>
         </dependency>
-        
         <dependency>
-            <groupId>javax.enterprise</groupId>
-            <artifactId>cdi-api</artifactId>
-            <version>1.2</version>
+            <groupId>javax</groupId>
```
Comment (johnament) [inline tck/pom.xml]: Can we avoid depending on the entirety of javaee?  I see it's done in a few places.

File: api/src/main/java/org/eclipse/microprofile/jwt/JWTCallerPrincipal.java (hunk 1 above)
Comment (starksm64) [inline api/src/main/java/org/eclipse/microprofile/jwt/JWTCallerPrincipal.java]: I'll move this to an ri subproject as reference implementation as I want a default implementation to illustrate the concepts.

File: api/src/test/resources/jwk-content1.json (hunk 3 above)
Comment (starksm64) [inline api/src/test/resources/jwk-content1.json]: I'm expecting that the security integration layer is what is handling the mapping from the target microservice based on it's CDI name and declared role requirements.

File: tck/pom.xml (hunk 4 above)
Comment (starksm64) [inline tck/pom.xml]: We can, I was just following what has been done elsewhere.

**Comment by starksm64** [comment]
This was signed in an amend, but apparently that does not meet the signing requirements, so I'll do a new pull request with changes for the comments.

**Pull request closed without merging**